packages/fluentui/react-bindings/src/styles/resolveStyles.ts

```diff
@@ -99,10 +124,11 @@ const resolveStyles = (
     }
   }
 
-  const componentCacheKey =
-    cacheEnabled && displayName && stylesProps
-      ? `${displayName}:${JSON.stringify(stylesProps)}${styleParam.rtl}${styleParam.disableAnimations}`
-      : '';
+  const propsCacheKey = cacheEnabled ? JSON.stringify(stylesProps) : '';
+  const variablesCacheKey = cacheEnabled && performance.enableBooleanVariablesCaching ? JSON.stringify(variables) : '';
```

Comment:
For the props it was not problematic that the props order would change, because we are always mapping them in the same order. On the other hand for the variables we cannot guarantee that, as those are purely users input. I propose to always order them alphabetically and than we can safely use `JSON.stringify`, or clearly state somewhere that those should always be in the same order...

Reply:
Added this to **Limitations** section in PR description